### PR TITLE
[script] [charge-holy-weapon] configurable holy power threshold when to recharge weapon

### PR DESCRIPTION
**Pull request by KatoakDR**
### Dependencies
* ~~Depends on https://github.com/rpherbig/dr-scripts/pull/4692~~

### Background
* [Madigan](https://discord.com/channels/745675889622384681/745678209449853049/796807722011721748) requested more fine-grained control over when the script chooses to recharge the holy weapon.
* Taking into consideration that some people have holy icons they can use anywhere, others may have to travel to an altar, and personal preferences, people may want to recharge their weapon at different thresholds. Similar to preferences on when to refil a car's gasoline... some do it at half tank, some quarter tank, some run out of gas ;)

### Changes
* Add the 8 power level match strings and metadata to `data/base-spells.yaml`
* Support new yaml setting `holy_weapon: power_level_threshold:` (optional, number in range 1-8)
* Refactor `check_charge_level` method to compare the holy power level seen when looking at the weapon to the character's configured threshold and decide whether to recharge or not.
* If unable to detect holy power on the weapon then exits with error message -- character may have specified a non-holy weapon.

## Tests

Madigan, again, has graciously [offered to help test](https://discord.com/channels/745675889622384681/745678209449853049/797368525315899424) the script updates. Awaiting their test results.

### skips when weapon is above threshold
```
>;charge-holy-weapon
--- Lich: charge-holy-weapon active.
[charge-holy-weapon]>wield my glaes.greatsword
You draw out your glaes greatsword from the claymore sheath, gripping it firmly in your right hand and balancing with your left.
>
[charge-holy-weapon]>look my glaes greatsword
Your greatsword is emanating a powerful sense of holiness.
 
The extra long hilt of this blade is capped with a heavy steel pommel designed to counter balance the length of the weapon.  Forming the protective crossguard, a pair of sinuous silver dragons, wings outstretched, mirror each other clasping a pristine white Therengian rose between the them.
>
[charge-holy-weapon: Detected holy power level of 7 with an estimated 1000 charges remaining]
[charge-holy-weapon: Your holy power level threshold is 3 or below]
| Above holy power level threshold, skipping
[charge-holy-weapon]>sheath my glaes.greatsword
You put your greatsword in your claymore sheath.
You sheathe the glaes greatsword in your claymore sheath.
>
--- Lich: charge-holy-weapon has exited.
```

### CODE REVIEW

**Comment by KatoakDR** [comment]
Converted to draft while wait on more test results.

**Comment by rpherbig** [comment]
Remind me to take a look at this once master is merged in and/or you get more test results.

**Comment by jayRyan24** [comment]
I think it's mostly fine, but I have some things to add:

1. Madigan doesn't really know anything about how the system works or how paladins work, he's fairly new. Adding a setting for a person that is inexperienced is kind of useless. 
2. The usefulness of this script was intended to be as a before/after or a constant check within a segment of T2. 200 remaining charges will make sure that you don't try to charge more than capable and cause bad things to happen. Unless you are farming, and not hunting for EXP, then you should never hit a point where the check of 200 swings is an issue. 
3. A simple 'force' argument could be used to bypass that logic and force a recharge if people are using it outside of the above intended condition.
4. If anything the charge levels are not a part of spell data.

File: charge-holy-weapon.lic [hunk 1]
```diff
@@ -26,24 +27,34 @@ class HolyWeapon
   end
 
   def check_charge_level(hw_settings)
-    @em.wield_weapon(hw_settings['weapon_name'])
+    weapon = hw_settings['weapon_name']
+    threshold = hw_settings['power_level_threshold'] || 3
```
Comment (jayRyan24) [inline charge-holy-weapon.lic:31]: Why default to 3 and not 4 like previous?

**Comment by KatoakDR** [comment]
Closed in favor of a simpler approach, a `force` argument

File: charge-holy-weapon.lic (hunk 1 above)
Comment (KatoakDR) [reply charge-holy-weapon.lic:31]: That was a typo, I thought 3 did refer to the the old power level value, thanks. It shouldn't have changed the behavior from the original script.

**Pull request closed without merging**